### PR TITLE
Support for PHP 8.4

### PR DESCRIPTION
**Pull request by PBXg33k**
This PR adds support for php 8.4. 

Code-wise i only had to address the deprecation notice it triggered on implicit nullable parameters.

My bad if the PR isn't targeted correctly. I read the contributing guidelines to late, if this bothers you i can re-fork and branch directly off the 5.0 branch.

### CODE REVIEW

File: src/Uri.php [hunk 1]
```diff
@@ -31,7 +31,7 @@ public function getScheme(): string
     private ?string $username = null;
     private ?string $password = null;
 
-    public function withUserInfo(string $user, string $password = null): UriInterface
+    public function withUserInfo(?string $user, ?string $password = null): UriInterface
```
Comment (byjg) [inline src/Uri.php:34]: The parameter $user in the [UriInterface](https://github.com/php-fig/http-message/blob/master/src/UriInterface.php#L208) doesn't accept null

```suggestion
    public function withUserInfo(string $user, ?string $password = null): UriInterface
```

File: composer.json [hunk 2]
```diff
@@ -10,12 +10,12 @@
   "prefer-stable": true,
   "minimum-stability": "dev",
   "require": {
-    "php": ">=8.1 <8.4",
+    "php": ">=8.1",
```
Comment (byjg) [inline composer.json:13]: It is great to include a limit version. 

```suggestion
    "php": ">=8.1 <8.5",
```

**Review by byjg — CHANGES_REQUESTED** [review]
Thank you for your PR. I am suggesting a few modifications.

File: src/Uri.php (hunk 1 above)
Comment (PBXg33k) [reply src/Uri.php]: Sharp observation, i overlooked that constraint :)

**Comment by PBXg33k** [comment]
Thanks for the feedback, addressed the issues you pointed out.

**Review by byjg — APPROVED** [review]

**Comment by byjg** [comment]
PR Merged and new release 5.0.3 provided.

**Pull request merged**